arch/arm/core/swap_helper.S

```diff
@@ -61,6 +61,8 @@ SECTION_FUNC(TEXT, __pendsv)
 #elif defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
     movs.n r0, #_EXC_IRQ_DEFAULT_PRIO
     msr BASEPRI, r0
+    dsb /* Make the effect of disabling interrupts be realized immediately */
```

Comment:
This has no harm, @agross-linaro but I am not 100% sure it is actually needed. That's because LDRs bellow only load the __current thread which i am not sure that it can be changed by a late ISR...